### ISSUE TITLE
ctsProfile check failed

### ISSUE THREAD

**Issue by abbradar** [kdrag0n/safetynet-fix#81]
Hi,

I tried both 2.0.0 and 2.1.0 (preview) versions of safetynet-fix -- I still can't pass ctsProfile. Using XPrivacyLua works, so I expect the problem is on safetynet-fix side.

**Comment by DavidBerdik** [comment]
How do you set up XPrivacyLua to bypass SafetyNet?

**Comment by abbradar** [comment]
One needs to limit `Use tracking` for Google Play Services. You need to install LSPosed though, and I'm not yet sure anything else wasn't broken by this. Still experimenting.

If there is any additional info wrt safetynet-fix I could provide, I'll be happy to help!

**Comment by DavidBerdik** [comment]
> One needs to limit `Use tracking` for Google Play Services. You need to install LSPosed though, and I'm not yet sure anything else wasn't broken by this. Still experimenting.

Interesting! Although I don't understand how limiting the `Use tracking` option for Google Play Services would bypass it. It sounds like a solution that, as you say, probably breaks other things, and is also highly likely to broken by Google in the future. I am looking for alternative solutions to this module for bypassing SafetyNet, as I am concerned about the direction this project is headed. I am worried about the fact that it seems to be becoming increasingly closed-source.

> If there is any additional info wrt safetynet-fix I could provide, I'll be happy to help!

I think that's all I need for now. Thank you! I've been using this module on the devices I manage since v1.0.2, and have never had any problems with it. All the devices that I maintain have been using v1.2.0 since it was released, and it's still working on my phone (Pixel 3 XL), however, on my parents' phones (Pixel 4a), the ctsProfile has started failing recently.

**Comment by dongerlel** [comment]
Just install the addon visible on the top of my screenshot:
![image](https://user-images.githubusercontent.com/12804761/132542948-eb373c87-cea8-4d8a-b9fe-dc9fa81bb32d.png)
![image](https://user-images.githubusercontent.com/12804761/132542991-b840dde8-dfd4-431e-97a4-7fdf49c47045.png)
cheers

**Comment by abbradar** [comment]
@dongerlel Can you share a link? I couldn't find it by full name.

Does anyone have luck with 2.1.0? Maybe I have something else missing on my device.

BTW I still needed `GPay SQLite Fix` addon for Google Pay to work.

**Comment by DavidBerdik** [comment]
@dongerlel I also have not been able to find it, but still looking.

**Comment by DavidBerdik** [comment]
Perhaps @windeqk would know since they are listed as the module developer?

**Comment by judemille** [comment]
Running a Pixel 5a here, I found that I could only get it to work with 1.2.0 and using MagiskHide Props Config to force basic attestation (so now some apps think i have a Nexus 5 lol)

**Comment by DavidBerdik** [comment]
> Running a Pixel 5a here, I found that I could only get it to work with 1.2.0 and using MagiskHide Props Config to force basic attestation (so now some apps think i have a Nexus 5 lol)

Well, that's an interesting method. I thought the MagiskHide Props Config module was entirely useless for bypassing SafetyNet at this point.

**Comment by Didgeridoohan** [comment]
> Running a Pixel 5a here, I found that I could only get it to work with 1.2.0 and using MagiskHide Props Config to force basic attestation (so now some apps think i have a Nexus 5 lol)

Don't pick the default... Choose from the list and you can at least make it seem like a regular Pixel 5.

> Well, that's an interesting method. I thought the MagiskHide Props Config module was entirely useless for bypassing SafetyNet at this point.

Using a certified device fingerprint is still quite necessary on custom ROMs (even though many developers now already include one in their ROM, but often annoyingly one that has nothing to do with the device in question). And changing model props is now also necessary on many devices, but v2.1.0 of USNF does that dynamically with the help of Riru.

**Comment by DavidBerdik** [comment]
> Using a certified device fingerprint is still quite necessary on custom ROMs (even though many developers now already include one in their ROM, but often annoyingly one that has nothing to do with the device in question). And changing model props is now also necessary on many devices (hence a Pixel 5a looking like a Nexus 5 above), but v2.1.0 of USNF does that dynamically with the help of Riru.

Ah alright. I'm still using stock Android, but will be moving to LineageOS in a few months once my beloved Pixel 3 XL reaches its EOL. 😞 I guess I will learn about the joys of device fingerprint problems at that point. I never had such problems the last time I used LineageOS. Of course, SafetyNet was a lot less nasty in those days.

**Comment by judemille** [comment]
> Don't pick the default... Choose from the list and you can at least make it seem like a regular Pixel 5.

I had to set the force basic attestation to a pixel 1 to work, since that was released before Android 8. I tried setting it to the 5 or 5a but it didn't pass

**Comment by username227** [comment]
I have been using the solution with XPrivacyLua for over a week now, and I have not noticed it breaking anything else. As long as it is working, I am going to leave it as is. But the moment it doesn't work any longer, I will be experimenting with this great idea for a program. I only wish somebody would be able to include something to bypass MagiskHide completely before it becomes completely obsolete.

**Comment by DavidBerdik** [comment]
> I only wish somebody would be able to include something to bypass MagiskHide completely before it becomes completely obsolete.

v2.1.0 of this module integrates the MagiskHide features directly in to it, however, it's currently not available without a Patreon subscription. - https://kdrag0n.dev/patreon/safetynet-fix/v2.1.0

**Comment by username227** [comment]
> > I only wish somebody would be able to include something to bypass MagiskHide completely before it becomes completely obsolete.
> 
> v2.1.0 of this module integrates the MagiskHide features directly in to it, however, it's currently not available without a Patreon subscription. - https://kdrag0n.dev/patreon/safetynet-fix/v2.1.0

Good too know, thanks. Is this going to remain a paid module or will it eventually become open source like these?

**Comment by DavidBerdik** [comment]
> Good to know, thanks. Is this going to remain a paid module or will it eventually become open source like these?

It will eventually become open source. The Patreon subscription just grants you early download access.

**Comment by DavidBerdik** [comment]
It looks like the hunt for @windeqk's mysterious Magisk module has been taken to Reddit, but as of my sharing, it hasn't been found, as the participants seem to not realize that it is not just the next version of this module: https://www.reddit.com/r/Magisk/comments/pluise

**Comment by username227** [comment]
> > Good to know, thanks. Is this going to remain a paid module or will it eventually become open source like these?
> 
> You're welcome! It will eventually become open source. The Patreon subscription just grants you early download access.

any idea what the usual timetable is for the open source release?

**Comment by DavidBerdik** [comment]
> any idea what the usual timetable is for the open source release?

Approximately one week according to @kdrag0n.

Source: https://github.com/kdrag0n/safetynet-fix/issues/87#issuecomment-917340843

**Comment by stamepicmorg** [comment]
> Perhaps @windeqk would know since they are listed as the module developer?

do you have any proofs that this account is same with those fake screenshot? it all looks like a lazy scam. imho.

**Comment by thelastbatman** [comment]
One week over. Still not released. So much greed for money in people

**Comment by Didgeridoohan** [comment]
Holy heck... The dude said that it usually takes about a week but that it depends on how much he's got on his plate at the time. Just wait, no need for snark.

**Comment by thelastbatman** [comment]
Dude firstly no one does this thing. people ask for donations and people donate according to will. Keep early releases for people who pay is not something nice.

**Comment by DavidBerdik** [comment]
> do you have any proofs that this account is same with those fake screenshot? it all looks like a lazy scam. imho.

Nope. I have no proof at all. The only reason I didn't initially assume anything malicious about it is because:

1. There is nothing to be gained from sharing a fake screenshot in this context. If someone starts trying to make money off of it, then that will change, but to my knowledge, nobody has attempted to do this so far.
2. This repository does not contain any code right now. It's just a fork of a template intended for being a foundation for building Magisk modules.

Assuming that this repo is going to eventually contain something, perhaps @windeqk can tell us what is going on?

I've been following a post on r/Magisk on Reddit asking about this module, but so far, it seems that nobody has found it: https://www.reddit.com/r/Magisk/comments/pluise

> One week over. Still not released.

Not to worry. It's only a matter of time before someone else picks this up. I'm sure we have all seen this kind of pattern before: a previously-open source project slowly gets locked up behind a paywall and eventually, someone else creates their own open source project without a paywall, thus eliminating the need for the initial project. Remember SuperSU?

**Comment by kdrag0n** [comment]
@thelastbatman Please stop commenting about ETAs here. As per my response to the issue you opened:

> I've already mentioned that it will be released **when I get a chance to do it.** That includes building for release, updating the readme, creating a GitHub release, updating the release info on my website, and writing a Telegram post.

**Comment by huxley101** [comment]
Seriously to anyone acting entitled demanding things, give kdrag0n a break. We need all the friends we can get in this game.

**Comment by osm0sis** [comment]
This should probably be closed now that v2.1.0 is out. Likely Play Services, Play Store and Pay just need to be force closed and have their app data fully cleared.

**Comment by judemille** [comment]
On 2.1.0, it's working flawlessly for me, disabled props config force basic without issue, thanks @kdrag0n for your work! I would donate if I had money lol. Keep up the good work

**Comment by woubulbus** [comment]
I can attest to it working. 2.1.0 works flawlessly so far. Lineageos 18.1 (v.20210911) on Moto ocean.

**Comment by xowex777** [comment]
wt., 14 wrz 2021, 23:43 użytkownik woubulbus ***@***.***>
napisał:

**Comment by dmitryuk** [comment]
Xiaomi mi 10T
The fix doesn't work. 
Installed Riru, MagicHide props config, safetynet-fix 2.1.0
Magist is hidden, name changed, cache of Google play services / Pay cleared.
What I should do next?
Thanks

**Comment by DavidBerdik** [comment]
Make sure that you have MagiskHide enabled in the settings panel of the Magisk app.

**Comment by kdrag0n** [comment]
Closing now that v2.1.1 is public and fixes have been suggested for continued SafetyNet failures.

**Comment by benwiley4000** [comment]
I have restarted and cleared storage for all play apps and this doesn't seem to have helped. I have installed Magisk, basic attestation for Google Pixel 3a (I have a 4a), safetynet fix installed, MagiskHide activated for Google Play services and all its activities.  Is there something I forgot ? Thanks

**Comment by DavidBerdik** [comment]
@benwiley4000 Magisk 24 was released a couple days ago, and because it has many significant changes related to how SafetyNet bypassing is done, there are lots of users validating their SafetyNet status, and some of the more popular validation apps have been hitting their daily request quotas, which means they are failing even if your device is passing. I ran in to this same issue today when updating to Magisk 24 and testing my SafetyNet state, but when I tried using a less popular validation app, I passed. Here is the one I would recommend using.

https://play.google.com/store/apps/details?id=com.atominvention.rootchecker

Here are the references I found when trying to figure out why I was failing today:
- https://developer.android.com/training/safetynet/attestation#quota-monitoring
- https://forum.xda-developers.com/t/discussion-magisk-the-age-of-zygisk.4393877/page-5#post-86337015
